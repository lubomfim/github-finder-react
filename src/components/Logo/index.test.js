import React from 'react';
import { render, screen } from '@testing-library/react';

import { Logo } from '.';
import { MemoryRouter } from 'react-router-dom';

const Component = (prop) => {
  return (
    <MemoryRouter>
      <Logo $home={prop ? prop : false} />
    </MemoryRouter>
  );
};

it('should render correctly', () => {
  const { asFragment } = render(Component());

  const link = screen.getByRole('link');
  const img = screen.getByRole('img', 'github Logo');

  expect(link).toBeInTheDocument();
  expect(img).toBeInTheDocument();
  expect(asFragment(Component())).toMatchSnapshot();
});

it('should render style correctly when condition is true', () => {
  const { container } = render(Component(true));
  expect(container).toMatchSnapshot();
});
