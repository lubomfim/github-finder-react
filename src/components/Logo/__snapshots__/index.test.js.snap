// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<DocumentFragment>
  .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding-top: 20px;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

.c1 {
  max-width: 35px;
}

<div
    class="c0"
  >
    <a
      class="c1"
      href="/"
    >
      <img
        alt="github logo"
        class=""
        src="images/logo/github.png"
      />
    </a>
  </div>
</DocumentFragment>
`;

exports[`should render style correctly when condition is true 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding-top: 50px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 {
  max-width: 50px;
}

<div>
  <div
    class="c0"
  >
    <a
      class="c1"
      href="/"
    >
      <img
        alt="github logo"
        class=""
        src="images/logo/github.png"
      />
    </a>
  </div>
</div>
`;
